src/components/overlays/SavePrompt.tsx

```diff
@@ -3,72 +3,158 @@ import _ from 'lodash';
 import { observer } from 'mobx-react';
 import React from 'react';
 import { FiArrowRight } from 'react-icons/fi';
-import { getLabel } from '~/codeLists/labelCodes';
+import {
+    linkPropertyCodeList,
+    nodePropertyCodeList,
+    routePropertyCodeList,
+    stopAreaPropertyCodeList,
+    stopPropertyCodeList
+} from '~/codeLists/propertyCodeList';
 import codeListStore from '~/stores/codeListStore';
+import { dateToDateString } from '~/util/dateFormatHelpers';
 import * as s from './savePrompt.scss';
 
-interface ISavePromptProps {
+type Model = 'node' | 'stop' | 'link' | 'route' | 'stopArea';
+
+interface ISaveModel {
+    model: Model;
     newData: Object;
-    oldData: Object;
-    type: string;
+    oldData: Object | null;
+}
+
+interface ISavePromptProps {
+    saveModels: ISaveModel[];
 }
 
-const callRenderByProperty = (property: string, oldLink: Object, newLink: Object) => {
-    switch (property) {
-        case 'geometry':
-            return renderChangeRow('Vanha geometria', 'Uusi geometria');
-        case 'municipalityCode':
-            return renderChangeRow(
-                codeListStore!.getCodeListLabel('Kunta (KELA)', oldLink[property]),
-                codeListStore!.getCodeListLabel('Kunta (KELA)', newLink[property])
-            );
+const PREVENTED_CHANGE_ROW_PROPERTIES = ['modifiedOn', 'modifiedBy'];
+
+const renderSaveModelSection = (saveModel: ISaveModel, key: string) => {
+    const newData = _.cloneDeep(saveModel.newData);
+    const oldData = _.cloneDeep(saveModel.oldData);
+    for (const property in newData) {
+        const newValue = newData[property];
+        const oldValue = oldData ? oldData[property] : '';
+        if (_.isEqual(newValue, oldValue)) {
+            delete newData[property];
+        }
+    }
+    return (
+        <div key={key}>
+            {Object.keys(newData).map((property: string, index: number) => {
+                if (PREVENTED_CHANGE_ROW_PROPERTIES.includes(property)) return null;
+
+                const propertyLabel = _getLabel(saveModel.model, property);
+                return (
+                    <div
+                        key={`${key}-${index}`}
+                        className={classnames(s.formItem, s.savePromptRow)}
+                    >
+                        <div className={s.inputLabel}>{propertyLabel}</div>
+                        {renderChangeRow(
+                            property,
+                            _getPropertyValue(saveModel.model, property, oldData, false),
+                            _getPropertyValue(saveModel.model, property, newData, true)
+                        )}
+                    </div>
+                );
+            })}
+        </div>
+    );
+};
+
+const _getLabel = (model: Model, property: string) => {
```

Comment:
What if we had 
```
const propertyCodeList = {
    node: nodePropertyCodeList,
    stop: stopPropertyCodeList,
    link: linkPropertyCodeList,
    route: routePropertyCodeList,
    stopArea: stopAreaPropertyCodeList
};

export { propertyCodeList };
```
in propertyCodeList.ts and then we could replace switch case here with something like
```
    const propertyCode = propertyCodeList[model][property];
    if (propertyCode) {
        return propertyCode;
    }
    throw `Unsupported model given for savePrompt: ${model}`;
```